src/app/lib/device/dlna.js

```diff
@@ -17,74 +17,124 @@
         makeID: makeID,
 
         initialize: function (attrs) {
-            this.device = attrs.device;
-            this.client = new MediaRendererClient(this.device.xml);
-            this.attributes.name = this.device.name;
-            this.attributes.address = this.device.host;
+            this.player = attrs.player;
+            this.attributes.name = this.player.name;
+            this.attributes.address = this.player.host;
         },
 
         play: function (streamModel) {
-            var metadata = {
-                title: 'Butter DLNA', // upnp-mediarendered-client object
-                type: 'video',
-                url: streamModel.get('src'),
-                protocolInfo: 'http-get:*:video/mp4:*'
-            };
-
-            if (streamModel.get('subFile')) { // inject subtitles
-                metadata.subtitlesUrl = 'http:' + metadata.url.split(':')[1] + ':9999/video.srt';
+            var url = streamModel.get('src');
+            var self = this;
+            var media;
+            var url_video = url;
+            var url_subtitle = 'http:' + url.split(':')[1] + ':9999/video.srt';
+            var metadata = null;
+            var subtitle = streamModel.get('subFile');
+            if (subtitle) {
+                media = {
+                    title: streamModel.get('title'),
+                    subtitles: ['http:' + url.split(':')[1] + ':9999/subtitle.vtt'],
```

Comment:
There also is a bug where subtitleserver is too slow and fails to send the subtitle in time, and you have to close and restart streaming in order to have a subtitle